editor/test/integration/editor_extensions_test.clj

```diff
@@ -375,20 +377,158 @@
       ;; all the changes should be reverted — a single transaction!
       (test-initial-state!))))
 
-(test-util/with-loaded-project "test/resources/editor_extensions/save_test"
-  (let [output (atom [])
-        _ (extensions/reload! project :all
-                              :reload-resources! (make-reload-resources-fn workspace)
-                              :display-output! #(swap! output conj [%1 %2])
-                              :save! (make-save-fn project))
-        handler+context (handler/active
-                          (:command (first (handler/realize-menu :editor.asset-browser/context-menu-end)))
-                          (handler/eval-contexts
-                            [(handler/->context :asset-browser {} (->StaticSelection []))]
-                            false)
-                          {})]
-    @(handler/run handler+context)
-    ;; see test.editor_script: it uses editor.transact() to set a file text, then reads
-    ;; the file text from file system, then saves, then reads it again.
-    (is (= [[:out "file read: before save = 'Initial text', after save = 'New text'"]]
-           @output))))
+(deftest save-test
+  (test-util/with-loaded-project "test/resources/editor_extensions/save_test"
+    (let [output (atom [])
+          _ (extensions/reload! project :all
+                                :reload-resources! (make-reload-resources-fn workspace)
+                                :display-output! #(swap! output conj [%1 %2])
+                                :save! (make-save-fn project))
+          handler+context (handler/active
+                            (:command (first (handler/realize-menu :editor.asset-browser/context-menu-end)))
+                            (handler/eval-contexts
+                              [(handler/->context :asset-browser {} (->StaticSelection []))]
+                              false)
+                            {})]
+      @(handler/run handler+context)
+      ;; see test.editor_script: it uses editor.transact() to set a file text, then reads
+      ;; the file text from file system, then saves, then reads it again.
+      (is (= [[:out "file read: before save = 'Initial text', after save = 'New text'"]]
+             @output)))))
+
+(deftest coercer-test
+  (let [vm (vm/make)
+        coerce #(coerce/coerce vm %1 (vm/->lua %2))]
+
+    (testing "enum"
+      (let [enum (coerce/enum 1 4 false "foo" :bar)]
+        (is (= 1 (coerce enum 1)))
+        (is (= false (coerce enum false)))
+        (is (= "foo" (coerce enum "foo")))
```

Comment:
We might want to check `identical?` instead of equality for strings that are declared as enums?

It would be nice if we enforced the property that we don't want individual copies of these strings marshaled from Lua.

Reply:
Yes!